### PR TITLE
new hardening.md page

### PR DESCRIPTION
**Pull request by eaudetcobello**
* move the manual hardening steps from cis-hardening.md to hardening.md.
* adds disa stig findings that need manual intervention
* rename sections
* improve grammar

### CODE REVIEW

**Comment by bschimke95** [comment]
Please resolve the conflict

File: docs/src/snap/howto/security/hardening.md [hunk 1]
```diff
@@ -0,0 +1,3429 @@
...
+
+#### Set event rate limits
+
+```{note}
+Configuring event rate limits requires the cluster administrator's input
+in assessing the hardware and workload specifications/requirements.
+```
+
+
+Create a configuration file with the [rate limits][] and place it under
```
Comment (nhennigan) [inline docs/src/snap/howto/security/hardening.md:74]: not linked correctly

File: docs/src/snap/howto/security/hardening.md [hunk 2]
```diff
@@ -0,0 +1,3429 @@
+# {{product}} hardening guide
+
+By default, {{product}} is designed to be compliant with all hardening
```
Comment (nhennigan) [inline docs/src/snap/howto/security/hardening.md:3]: Contradiction with the line below "However, some hardening configurations are not applied by default" . In the CIS page we are clear that we comply with most, not all and our reasons why. "Out of the box Canonical Kubernetes complies with the majority of the recommended CIS security configurations."  We must be careful what we claim to be compliant with.

File: docs/src/snap/howto/cis-hardening.md [hunk 3]
```diff
@@ -22,224 +22,11 @@ This guide assumes the following:
 - You have root or sudo access to the machine
 
 
-## Post-deployment extra hardening steps
-
-The following hardening configurations are not part of the default {{product}}
```
Comment (nhennigan) [inline docs/src/snap/howto/cis-hardening.md:27]: I believe that this section is better suited remaining in this page. The purpose of this how to guide is to become cis compliant and get kube-bench to pass without warnings. Jumping from page to page may be confusing.

File: docs/src/snap/howto/security/hardening.md [hunk 4]
```diff
@@ -0,0 +1,3429 @@
...
+*	Affect compatibility with workloads.
+*	Require input from the administrator.
+
+Please evaluate the implications of each configuration before applying it.
+
+## Post-Deployment Configuration Steps
+
+The following configurations, when applied, will enhance your Kubernetes
+cluster’s security posture. After completing these steps, your cluster will be
+fully compliant with both DISA STIG recommendations and CIS hardening
```
Comment (nhennigan) [inline docs/src/snap/howto/security/hardening.md:22]: Again fully compliant I would take to mean no warnings whatsoever. These checks will only remove failing tests

File: docs/src/snap/howto/security/hardening.md [hunk 5]
```diff
@@ -0,0 +1,3429 @@
...
+administrator intervention. These steps are outlined in the [Post-Deployment
+Configuration Steps section][]. These are omitted by default because they may:
+
+* Impact performance.
+*	Affect compatibility with workloads.
+*	Require input from the administrator.
+
+Please evaluate the implications of each configuration before applying it.
+
+## Post-Deployment Configuration Steps
```
Comment (nhennigan) [inline docs/src/snap/howto/security/hardening.md:18]: I would consider renaming this section to better reflect what these steps do. All the hardening steps are post-deployment. These steps are the highest priority/ ones to remove failures from CIS and DISA STIG

File: docs/src/snap/howto/security/hardening.md [hunk 6]
```diff
@@ -0,0 +1,3429 @@
...
+
+However, some hardening configurations are not applied by default and require
+administrator intervention. These steps are outlined in the [Post-Deployment
+Configuration Steps section][]. These are omitted by default because they may:
+
+* Impact performance.
+*	Affect compatibility with workloads.
+*	Require input from the administrator.
+
+Please evaluate the implications of each configuration before applying it.
```
Comment (nhennigan) [inline docs/src/snap/howto/security/hardening.md:16]: I would also mention the manual steps at the bottom. This how-to has both the bare minimum to pass the tests and also a more comprehensive list of manual tests

File: docs/src/snap/howto/cis-hardening.md (hunk 3 above)
Comment (nhennigan) [reply docs/src/snap/howto/cis-hardening.md:27]: We can use an {include} statement if you want to reduce maintenance having it in two locations

**Review by nhennigan — COMMENTED** [review]
You need to add security to docs/src/howto/index.md for it to show in the navigation on the side.

Also please address spelling mistakes highlighted in the github runner.

There are several of the DISA STIG recommendations that are not mapped to the CIS controls. E.g V-242457: The Kubernetes kubelet config must be owned by root. -> Control 4.1.6. Is there a reason for this?

File: docs/src/snap/howto/cis-hardening.md [hunk 7]
```diff
@@ -22,224 +22,11 @@ This guide assumes the following:
...
-```
-chmod 600 /etc/systemd/system/snap.k8s.kubelet.service
-```
-
-Restart `kubelet`.
-
-```
-sudo systemctl restart snap.k8s.kubelet
-```
+## Post-Deployment Configuration Steps
```
[the remark this answers is not in the document]
Comment (eaudetcobello) [reply docs/src/snap/howto/cis-hardening.md:25]: TIL

File: docs/src/snap/howto/security/disa-stig-hardening.md [hunk 8]
```diff
@@ -0,0 +1,234 @@
+# DISA STIG for {{product}}
+
+Security Technical Implementation Guides (STIG) are developed by the Defense
+Information System Agency (DISA) for the U.S. Department of Defense (DoD).
+
+The Kubernetes STIGs contain guidelines on how to check remediate various
```
Comment (nhennigan) [inline docs/src/snap/howto/security/disa-stig-hardening.md:6]: to check and remediate

File: docs/src/snap/howto/security/disa-stig-hardening.md [hunk 9]
```diff
@@ -0,0 +1,234 @@
...
+For manual audits of DISA STIG hardening recommendations, please visit the
+[Comprehensive Hardening Checklist][].
+
+
+<!-- Links -->
+[Hardening]:security/hardening.md
+[Center for Internet Security (CIS)]:https://www.cisecurity.org/
+[kube-bench]:https://aquasecurity.github.io/kube-bench/v0.6.15/
+[CIS Kubernetes Benchmark]:https://www.cisecurity.org/benchmark/kubernetes
+[getting started]: ../tutorial/getting-started
```
Comment (nhennigan) [inline docs/src/snap/howto/security/disa-stig-hardening.md:230]: ../../tutorial/getting-started

File: docs/src/snap/howto/security/cis-hardening.md [hunk 10]
```diff
@@ -0,0 +1,152 @@
...
+For manual audits of CIS hardening recommendations, please visit the
+[Comprehensive Hardening Checklist][].
+
+
+<!-- Links -->
+[Hardening]:security/hardening.md
+[Center for Internet Security (CIS)]:https://www.cisecurity.org/
+[kube-bench]:https://aquasecurity.github.io/kube-bench/v0.6.15/
+[CIS Kubernetes Benchmark]:https://www.cisecurity.org/benchmark/kubernetes
+[getting started]: ../tutorial/getting-started
```
Comment (nhennigan) [inline docs/src/snap/howto/security/cis-hardening.md:149]: ../tutorial/getting-started

File: docs/src/snap/howto/security/cis-hardening.md [hunk 11]
```diff
@@ -0,0 +1,152 @@
...
+
+This guide assumes the following:
+
+- You have a bootstrapped {{product}} cluster (see the [getting started] guide)
+- You have root or sudo access to the machine
+
+
+## Critical post-deployment hardening steps
+
+By completing these steps, you can ensure your cluster achieves does not fail
```
Comment (nhennigan) [inline docs/src/snap/howto/security/cis-hardening.md:27]: you can ensure your cluster ~~achieves~~ does not fail

File: docs/src/snap/howto/security/cis-hardening.md [hunk 12]
```diff
@@ -0,0 +1,152 @@
...
+
+<!-- Links -->
+[Hardening]:security/hardening.md
+[Center for Internet Security (CIS)]:https://www.cisecurity.org/
+[kube-bench]:https://aquasecurity.github.io/kube-bench/v0.6.15/
+[CIS Kubernetes Benchmark]:https://www.cisecurity.org/benchmark/kubernetes
+[getting started]: ../tutorial/getting-started
+[kube-bench release]: https://github.com/aquasecurity/kube-bench/releases
+[Post-Deployment Configuration Steps]: security/hardening.md#post-deployment-configuration-steps
+[Comprehensive Hardening Checklist]: security/hardening.md#comprehensive-hardening-checklist
```
Comment (nhennigan) [inline docs/src/snap/howto/security/cis-hardening.md:152]: hardening.md#comprehensive-hardening-checklist

File: docs/src/snap/howto/security/disa-stig-hardening.md [hunk 13]
```diff
@@ -0,0 +1,234 @@
...
+This guide assumes the following:
+
+- You have a bootstrapped {{product}} cluster (see the [getting started] guide)
+- You have root or sudo access to the machine
+
+
+## Critical post-deployment hardening steps
+
+By completing these steps, you can ensure your cluster achieves does not fail
+any of the DISA STIG hardening recommendations.
```
Comment (nhennigan) [inline docs/src/snap/howto/security/disa-stig-hardening.md:26]: V-242384 V-242385 V-242402 V-242403 V-242461 V-242462 V-242463 V-242464 V-242465

**Review by nhennigan — APPROVED** [review]

**Pull request merged**